src/tcgwars/logic/impl/gen4/SecretWonders.groovy

```diff
@@ -341,18 +341,33 @@ public enum SecretWonders implements LogicCardInfo {
           pokePower "Telepass", {
             text "Once during your turn , you may search your opponent’s discard pile for a Supporter card and use the effect of that card as the effect of this power. (This Supporter card remains in you opponent’s discard pile.) You can’t use more than 1 Telepass Poké-Power each turn. This power can’t be used if Gardevoir is affected by a Special Condition."
             actionA {
-            }
-          }
+              checkLastTurn()
+              assert bg.em().retrieveObject("Telepass") != bg.turnCount : "You can't use more than 1 Telepass Poke-Power each turn."
+              assert opp.discard.hasType(SUPPORTER) : "Your opponent has no supporters discarded."
+              if(opp.discard.hasType(SUPPORTER)){
+                powerUsed()
+                def card = opp.discard.select("Opponent's discard. Select a supporter.", cardTypeFilter(SUPPORTER)).first()
+                bg.deterministicCurrentThreadPlayerType=bg.currentTurn
+                bg.em().run(new PlayTrainer(card).setDontDiscard(true))
+                bg.clearDeterministicCurrentThreadPlayerType()
+              }  
+            } 
+          } 
           move "Psychic Lock", {
             text "60 damage. During your opponent’s next turn, your opponent can’t use any Poké-Powers on his or her Pokémon."
             energyCost P, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 60
+              getterA (IS_ABILITY_BLOCKED) { Holder h ->
+              if (h.effect.ability instanceof PokePower) {
```

Comment:
Could you indent the `if` block over to the right by one tab?

Reply:
fixed